Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/SurveyUrlProvider.kt

```diff
@@ -0,0 +1,42 @@
+package de.rki.coronawarnapp.datadonation.survey
+
+import de.rki.coronawarnapp.appconfig.AppConfigProvider
+import de.rki.coronawarnapp.datadonation.storage.OTPRepository
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class SurveyUrlProvider @Inject constructor(
+    private val appConfigProvider: AppConfigProvider,
+    private val otpRepo: OTPRepository
+) {
+
+    /**
+     * Provides an Url for the Data Donation Surveys.
+     *
+     * @throws IllegalStateException If no one time password (otp) can be loaded from the OTPRepository of if the
+     * AppConfig doesn't contain a link to a survey
+     */
+    suspend fun provideUrl(type: Surveys.Type): String {
```

Comment:
please consider to put this logic/method to _Surveys.kt_
we could then validate if the OTP has been used already this month.
If no OTP is there, we can just generate one...

Reply:
My plan was that after the OTP generation/validation/authentication (Part of Chris' subtask) ,`Surveys.kt` can simply call `provideUrl()`. At this point, a generated, validated and authenticated OTP should be stored in our `OtpRepository` that is then loaded by `SurveyUrlProvider` to assemble the URL.

I pushed a new commit where the `UrlProvider` is called in `Surveys.kt`. 

Do you think the `UrlProvider` shouldn't be a separate component but a method in `Surveys.kt`? What would be the benefit?

Reply:
The SurveyUrlProvider could be irritating if one calls it from another class than Surveys.
For e.g. one tries to get an URL for a High_Risk_Survey. He could use the url provider instead of surveys and so accidentally skip the generation/validation/authentication of the OTP.

Reply:
Is the OTP (re)generated on-demand?

What is the calling classes expected to do when `provideUrl` throws `IllegalStateException`?

Reply:
@BMItter 
I added a new commit. Now you need to pass the `OTP` to the `UrlProvider` and since only the Surveys class is going to generate `OTP`s, it shouldn't be called from anywhere else.

@d4rken 
I changed the UrlProvider so that we now already have to pass an authenticated `OTP` to it. 
I guess the calling class should show a "Something went wrong...." error message to the user in this case.